### PR TITLE
add: CRUD functionality for posts

### PR DESCRIPTION
**Pull request by ApurvaSawant11**
In this PR, I have added the below components and features:
- components/followBar
- components/modal
- features/home
- features/postSlice
- features/userProfile/userSlice

Please review and suggest

### CODE REVIEW

File: src/features/home/postSlice.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+};
+
+export const getAllPosts = createAsyncThunk(
+  "post/getAllPosts",
+  async (_, { rejectWithValue }) => {
+    try {
+      const { data } = await getAllPostsService();
+      return data;
+    } catch (error) {
+      return rejectWithValue(error);
```
Comment (b0llu) [inline src/features/home/postSlice.js:22]: Liked that you are using rejectWithValue

**Review by b0llu — APPROVED** [review]
Really well written code. Was not able to find anything to suggest. Great work!

File: src/components/followBar/FollowBar.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  return (
+    <div className="w-72 ml-8 p-4 sticky top-20 bg-white rounded-md h-max hidden md:block shrink-0">
+      <h2 className="text-md font-bold">Who to follow</h2>
+      {userSuggestions.map((user) => (
+        <div className="flex justify-between items-center mt-4">
+          <div
+            className="flex items-center gap-2 cursor-pointer"
+            onClick={() => navigate(`/${user.userHandle}`)}
+          >
+            <img src={user.profilePic} className="h-10 w-10 rounded-full" />
```
Comment (hsnice16) [inline src/components/followBar/FollowBar.jsx:33]: Is there any specific reason for not using `alt` here?

**Review by hsnice16 — APPROVED** [review]
Great work with the code. Couldn't find much to suggest. Good Job.

File: src/components/followBar/FollowBar.jsx (hunk 2 above)
Comment (ApurvaSawant11) [reply src/components/followBar/FollowBar.jsx:33]: No, must have missed it. I will add it in the next PR. Thanks

**Pull request merged**